src/util/user-home-dir.js

```diff
@@ -4,7 +4,8 @@ import ROOT_USER from './root-user.js';
 
 const path = require('path');
 
-const userHomeDir =
-  process.platform === 'linux' && ROOT_USER ? path.resolve('/usr/local/share') : require('os').homedir();
+export const home = require('os').homedir();
+
+const userHomeDir = process.platform === 'linux' && ROOT_USER ? path.resolve('/usr/local/share') : home;
```

Comment:
Hmm, I'm not sure  I fully understand the question. Would you mind explaining a bit? Windows is already filtered out in the ROOT_USER  here https://github.com/yarnpkg/yarn/blob/7a053e2ca07d19b2e2eebeeb0c27edaacfd67904/src/util/root-user.js#L4